core/commands/remotepin.go

```diff
@@ -0,0 +1,515 @@
+	},
+	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		psCh, errCh, err := lsRemote(ctx, req, env)
+		if err != nil {
+			return err
+		}
+
+		for ps := range psCh {
+			if err := res.Emit(&RemotePinOutput{
+				RequestID: ps.GetRequestId(),
+				Name:      ps.GetPin().GetName(),
+				Delegates: multiaddrsToStrings(ps.GetDelegates()),
+				Status:    ps.GetStatus().String(),
+				Cid:       ps.GetPin().GetCid().String(),
+			}); err != nil {
+				return err
+			}
+		}
+
+		return <-errCh
+	},
+	Type: RemotePinOutput{},
+	Encoders: cmds.EncoderMap{
+		cmds.Text: cmds.MakeTypedEncoder(func(req *cmds.Request, w io.Writer, out *RemotePinOutput) error {
+			fmt.Printf("pin_id=%v\n", out.RequestID)
+			fmt.Printf("pin_name=%q\n", out.Name)
+			for _, d := range out.Delegates {
+				fmt.Printf("pin_delegate=%v\n", d)
+			}
+			fmt.Printf("pin_status=%v\n", out.Status)
+			fmt.Printf("pin_cid=%v\n", out.Cid)
+			return nil
+		}),
+	},
+}
+
+func lsRemote(ctx context.Context, req *cmds.Request, env cmds.Environment) (chan pinclient.PinStatusGetter, chan error, error) {
+	opts := []pinclient.LsOption{}
+	if name, nameFound := req.Options[pinNameOptionName].(string); nameFound {
+		opts = append(opts, pinclient.PinOpts.FilterName(name))
+	}
+	if cidsRaw, cidsFound := req.Options[pinCIDsOptionName].([]string); cidsFound {
+		parsedCIDs := []cid.Cid{}
+		for _, rawCID := range cidsRaw {
+			parsedCID, err := cid.Decode(rawCID)
+			if err != nil {
+				return nil, nil, fmt.Errorf("CID %s cannot be parsed (%v)", rawCID, err)
+			}
+			parsedCIDs = append(parsedCIDs, parsedCID)
+		}
+		opts = append(opts, pinclient.PinOpts.FilterCIDs(parsedCIDs...))
+	}
+	if statusRaw, statusFound := req.Options[pinStatusOptionName].([]string); statusFound {
+		parsedStatuses := []pinclient.Status{}
+		for _, rawStatus := range statusRaw {
+			if pinclient.Status(rawStatus).String() == string(pinclient.StatusUnknown) {
+				return nil, nil, fmt.Errorf("status %s is not valid", rawStatus)
+			}
+			parsedStatuses = append(parsedStatuses, pinclient.Status(rawStatus))
+		}
+		opts = append(opts, pinclient.PinOpts.FilterStatus(parsedStatuses...))
+	}
+
+	service, _ := req.Options[pinServiceNameOptionName].(string)
+	url, key, err := getRemotePinServiceOrEnv(env, service)
+	if err != nil {
+		return nil, nil, err
+	}
+	c := pinclient.NewClient(url, key)
+
+	psCh, errCh := c.Ls(ctx, opts...)
+
+	return psCh, errCh, nil
+}
+
+var rmRemotePinCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Remove pinned objects from remote pinning service.",
+		ShortDescription: `
+Removes the pin from the given object allowing it to be garbage
+collected if needed.
+`,
+	},
+
+	Arguments: []cmds.Argument{
+		cmds.StringArg("request-id", false, true, "Request ID of the pin to be removed.").EnableStdin(),
```

Comment:
That's what the "false" flag is saying. It's not mandatory.

Reply:
What is the distinction between optional arguments and options, I guess?

Reply:
IIUC the difference is whether it the argument needs to be defined to be used.

e.g. if we had 3 things to be configured called a, b, c. We could do `command [a] [b] [c]` or `command --a=<val> --b=<val> --c=<val>`.

The question here is whether we should consider the request-id special enough to not need to name it, or whether we should just name everything. My feeling is it would be consistent to have them all be flags, but could be convinced otherwise

Reply:
My rationale was this: flags like --name --status etc, those used for matching essentially, refer to properties of a pin that don't unique identify it. An ID is unique. A command either uses an ID or flags, but never both. So, they are not the same "type" of thing. Also, using id as an argument to remove enables us later to support piping IDs into "rm" (you can't pipe into a flag)

Reply:
I'm leaning towards @petar opinion: "filtering" by `requestid` is a special case, at least in the current implementation. In `pin remote ls <requestid>` it actually does not do any filtering, uses different endpoint behind the scenes. Due to this, requires direct match, so it ignores parameters. Makes sense to keep it as argument, just to indicate that this is the primary identifier.

Reply:
I guess since we're requiring (or highly recommending) requestID be unique and not settable by the user that it can be special. How would you feel about having the same syntax for local pinning?

Also, if we're not going to make this a separate command (e.g. ipfs remote pin status <request ID>) then we should either check the filters before we return or error if they're set. Right now we ignore them if the request ID is set.